go/pkgbin/exec.go

```diff
@@ -0,0 +1,47 @@
+// pkgbin implements functionality related to Flow's packaging,
+// including finding binaries and artifcats which are part of
+// a packaged Flow release.
+package pkgbin
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Locate the full path of a binary by:
+// * Looking first for a binary that exist alongside the currently
+//   executing binary, in a common directory.
+// * Only if that's not found, then attempt to find a binary on the $PATH.
```

Comment:
This is nice, and makes me wonder if we should do the same thing in Rust. Definitely doesn't sound _necessary_ though, so IMO we just merge this and consider adding that later if we think of a reason to. The only thing that comes to mind at the moment is that `cargo run -p flowctl` wouldn't be able to find executables, but that's not something I imagine wanting to do as part of my workflow.